Add containerized proxy installation (#5097)
theforeman/foreman-documentation: guides/common/modules/proc_deploying-a-custom-ssl-certificate-to-smart-proxy-server.adoc

:_mod-docs-content-type: PROCEDURE

> :containerized:

Reviewer: ```suggestion
```

Please don't.

Author: Noticed it just a little afterwards, sorry!

= Deploying a custom SSL certificate to {SmartProxyServer}

[role="_abstract"]

> If your {ProjectServer} is configured with a custom certificate, you must configure your {SmartProxyServer} with a custom SSL certificate signed by a Certificate Authority.

Reviewer 2: ```suggestion
If your {ProjectServer} is configured with a custom certificate, you must configure your {SmartProxyServer} with a custom SSL certificate signed by the same Certificate Authority.
```

IMHO we can be more explicit here

ifndef::orcharhino[]
For more information, see xref:opening-required-ports[].
endif::[]
ifdef::containerized[]
* {SmartProxyServer} packages are installed.
For more information, see xref:installing-{smart-proxy-context}-packages[].
endif::[]

.Procedure
ifdef::containerized[]
. On your {ProjectServer}, generate an authentication bundle for the {SmartProxyServer}:
+
[options="nowrap" subs="+quotes,attributes"]
----

> # {foremanctl} auth-bundle _{smartproxy-example-com}_

Reviewer: ```suggestion
# {foremanctl} auth-bundle _{smartproxy-example-com}_ \
```

--certificate-source custom_server \
--certificate-server-certificate "_/root/{project-context}_cert/{project-context}_cert.pem_" \

> --certificate-server-key "_/root/{project-context}_cert/{project-context}_cert_key.pem_" \

Reviewer 2: I would have expected `project-context` to resolve to `smart-proxy` here, but the rendering shows `foreman` -- that's wrong

Reviewer 2: Ooooh, just grepped around, the others are using `{cert-name}` which uses different contextes:
```
% git grep :cert-name:
guides/common/assembly_configuring-satellite-custom-server-certificate.adoc::cert-name: {project-context}
guides/common/assembly_configuring-smart-proxy-servers-with-custom-ssl-certificates-for-load-balancing-with-puppet.adoc::cert-name: {smart-proxy-context}
guides/common/assembly_configuring-smart-proxy-servers-with-custom-ssl-certificates-for-load-balancing-without-puppet.adoc::cert-name: {smart-proxy-context}
guides/common/assembly_installing-smart-proxy-server.adoc::cert-name: {smart-proxy-context}
```

Reviewer 2: Or directly `smart-proxy-context` as the context here is clearly the proxy

Author: ```suggestion
--certificate-server-certificate "_/root/{project-context}_cert/{project-context}_cert.pem_" \
--certificate-server-key "_/root/{project-context}_cert/{smart-proxy-context}_cert_key.pem_" \
```

> --certificate-server-key "_/root/{project-context}_cert/{smart-proxy-context}_cert_key.pem_" \

Reviewer 2: ```suggestion
--certificate-server-certificate "_/root/{smart-proxy-context}_cert/{smart-proxy-context}_cert.pem_" \
--certificate-server-key "_/root/{smart-proxy-context}_cert/{smart-proxy-context}_cert_key.pem_" \
```

much more smart needed

> --certificate-server-key "_/root/{smart-proxy-context}_cert/{smart-proxy-context}_cert_key.pem_" \

Reviewer: ```suggestion
--certificate-server-key "_/root/{smart-proxy-context}_cert/{smart-proxy-context}_cert_key.pem_"
```

----
+
--
The options used in the command include the following:

`--certificate-source custom_server`:: Sets the certificate source to custom server certificates provided by the user.
`--certificate-server-certificate`:: Path to {ProjectServer} certificate file that is signed by a Certificate Authority.

> `--certificate-server-key`:: Path to the private key for the {ProjectServer} certificate.

Reviewer 2: ```suggestion
`--certificate-server-certificate`:: Path to {SmartProxyServer} certificate file that is signed by a Certificate Authority.
`--certificate-server-key`:: Path to the private key for the {SmartProxyServer} certificate.
```

Those are the ones you created in "Creating a custom SSL certificate for Smart Proxy server"

ifndef::orcharhino[]
For more information, see xref:opening-required-ports[].
endif::[]
ifdef::containerized[]
* {SmartProxyServer} packages are installed.
For more information, see xref:installing-{smart-proxy-context}-packages[].
endif::[]

.Procedure
ifdef::containerized[]
. On your {ProjectServer}, generate an authentication bundle for the {SmartProxyServer}:
+
[options="nowrap" subs="+quotes,attributes"]
----

> # {foremanctl} auth-bundle _{smartproxy-example-com}_

Reviewer 2: ```suggestion
# {foremanctl} auth-bundle --certificate-source custom_server --certificate-server-key proxy.key --certificate-server-certificate proxy.crt _{smartproxy-example-com}_
```

Author: Moved the options in [90ad11e](https://github.com/theforeman/foreman-documentation/pull/5097/commits/90ad11e8510f0b3887e13fb270db9aba6680c85b).

Just wanted to double check - the replaceable values can/should stay the same, or did you mean to change them as well?

Reviewer 2: the replaceable values were fine, I am just a lazy bum at typing ;)

Author: I don't think there's a polite way of saying "that's what I thought", so I'll just leave it at that :}

Reviewer 2: permission to be honest and impolite granted :)

----
. On your {ProjectServer}, copy the authentication bundle to {SmartProxyServer}:
+
[options="nowrap" subs="+quotes,attributes"]
----
# scp /var/lib/foremanctl/certs/bundles/_{smartproxy-example-com}_.tar.gz \
root@_{smartproxy-example-com}_:/root/_{smartproxy-example-com}_.tar.gz
----
+
[IMPORTANT]
====
The authentication bundle contains TLS private keys and OAuth credentials.
Treat the bundle as highly sensitive and delete any copies after deployment.
====
. On the {SmartProxyServer}, deploy the configuration:
+
[options="nowrap" subs="+quotes,attributes"]
----
# {foremanctl} deploy-proxy \
--flavor {foremanctl-proxy-scenario} \
--auth-bundle /root/_{smartproxy-example-com}_.tar.gz \
--foreman-fqdn _{foreman-example-com}_
--certificate-source custom_server \
--certificate-server-certificate "_/root/{project-context}_cert/{project-context}_cert.pem_" \
--certificate-server-key "_/root/{project-context}_cert/{project-context}_cert_key.pem_" \

> --certificate-server-ca-certificate "_/root/{project-context}_cert/ca_cert_bundle.pem_"

Reviewer 2: ```suggestion
```

----
+
--
The options used in the command include the following:

`--certificate-source custom_server`:: Sets the certificate source to custom server certificates provided by the user.
`--certificate-server-certificate`:: Path to {ProjectServer} certificate file that is signed by a Certificate Authority.
`--certificate-server-key`:: Path to the private key for the {ProjectServer} certificate.

> `--certificate-server-ca-certificate`:: Path to the Certificate Authority bundle.

Reviewer 2: we don't accept `ca-certificate`, it has to be the same as for the main server.

the rest you must place at `auth-bundle` time

> To regenerate certificates for an existing {SmartProxyServer}, add the `--certificate-renew` option to the `{foremanctl} deploy-proxy` command.

Reviewer 2: ```suggestion
To regenerate the bundle for an existing {SmartProxyServer}, add the `--certificate-renew` option to the `{foremanctl} auth-bundle` command.
```
